doc/design/api.md

```diff
@@ -0,0 +1,90 @@
+# Design Doc: PaddlePaddle API
+
+## Ingredients
+
+As the first step of our design, we list important concepts in deep
+learning and try to figure their relationship, as shown below:
+
+```
+Model = {topology, parameters}
```

Comment:
should we add a concept of Cost, there are several kind of cost, and different cost may compose to a new kind of Cost.